### ISSUE TITLE
from . import _swigfaiss ImportError: DLL load failed: 找不到指定的模块。

### ISSUE THREAD

**Issue by Nandu-H** [facebookresearch/faiss#1942]
# Summary

I met a strenge problem, I try hardly to install faiss-cpu on my win10,but it failed again and again.........
Was there anyone met this before?

# Platform

OS:  Windows 10

Faiss version: faiss-cpu                 1.7.1           py3.7_h2272212_1_cpu    pytorch
Installed from:  anaconda from install.md

Faiss compilation options: The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - pytorch/win-64::faiss-cpu==1.7.1=py3.7_h2272212_1_cpu
  - defaults/win-64::mkl_fft==1.3.0=py37h46781fe_0
  - defaults/win-64::mkl_random==1.0.4=py37h343c172_0
  - defaults/win-64::numpy==1.17.0=py37h19fb1c0_0
  but it don't work

Running on:
- [ -] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [ -] Python

# Reproduction instructions

 File "e:/BaiduNetdiskDownload/jingdong_NLP/lecture_code/Assignment3-1/Assignment3-1/retrieval/hnsw_faiss.py", line 23, in <module>
    import faiss
  File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\__init__.py", line 18, in <module>
    from .loader import *
  File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\loader.py", line 65, in <module>
    from .swigfaiss import *
  File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\swigfaiss.py", line 13, in <module>
    from . import _swigfaiss
ImportError: DLL load failed: 找不到指定的模块。

**Comment by nithinreddyy** [comment]
Same here 
**ImportError: DLL load failed: The specified module could not be found.**
Can anyone please help with this issue. I'm using python 3.7.9 with pip environment

**Comment by nithinreddyy** [comment]
> # Summary
> I met a strenge problem, I try hardly to install faiss-cpu on my win10,but it failed again and again.........
> Was there anyone met this before?
> 
> # Platform
> OS: Windows 10
> 
> Faiss version: faiss-cpu 1.7.1 py3.7_h2272212_1_cpu pytorch
> Installed from: anaconda from install.md
> 
> Faiss compilation options: The environment is inconsistent, please check the package plan carefully
> The following packages are causing the inconsistency:
> 
> * pytorch/win-64::faiss-cpu==1.7.1=py3.7_h2272212_1_cpu
> * defaults/win-64::mkl_fft==1.3.0=py37h46781fe_0
> * defaults/win-64::mkl_random==1.0.4=py37h343c172_0
> * defaults/win-64::numpy==1.17.0=py37h19fb1c0_0
>   but it don't work
> 
> Running on:
> 
> * [ -] CPU
> * [ ]  GPU
> 
> Interface:
> 
> * [ ]  C++
> * [ -] Python
> 
> # Reproduction instructions
> File "e:/BaiduNetdiskDownload/jingdong_NLP/lecture_code/Assignment3-1/Assignment3-1/retrieval/hnsw_faiss.py", line 23, in
> ```
> import faiss
> ```
> 
> File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss__init__.py", line 18, in
> from .loader import *
> File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\loader.py", line 65, in
> from .swigfaiss import *
> File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\swigfaiss.py", line 13, in
> from . import _swigfaiss
> ImportError: DLL load failed: 找不到指定的模块。

pip uninstall faiss-cpu

Install new version

pip install faiss-cpu

It will fix the error now

**Comment by mdouze** [comment]
Seems to be fixed. Closing.

**Comment by fingoldo** [comment]
Hi, it's not fixed (as of faiss_cpu-1.7.1.post2-cp38-cp38-win_amd64.whl), and it's a big problem know on forums like StackOverflow. It's preventing windows users from working with faiss, unfortunately.
```python

C:\ProgramData\Anaconda3\lib\site-packages\faiss\swigfaiss.py in <module>
     11 # Import the low-level C/C++ module
     12 if __package__ or "." in __name__:
---> 13     from . import _swigfaiss
     14 else:
     15     import _swigfaiss

ImportError: DLL load failed while importing _swigfaiss: The specified module could not be found.
```

**Comment by Nandu-H** [comment]
> > # Summary
> > I met a strenge problem, I try hardly to install faiss-cpu on my win10,but it failed again and again.........
> > Was there anyone met this before?
> > # Platform
> > OS: Windows 10
> > Faiss version: faiss-cpu 1.7.1 py3.7_h2272212_1_cpu pytorch
> > Installed from: anaconda from install.md
> > Faiss compilation options: The environment is inconsistent, please check the package plan carefully
> > The following packages are causing the inconsistency:
> > 
> > * pytorch/win-64::faiss-cpu==1.7.1=py3.7_h2272212_1_cpu
> > * defaults/win-64::mkl_fft==1.3.0=py37h46781fe_0
> > * defaults/win-64::mkl_random==1.0.4=py37h343c172_0
> > * defaults/win-64::numpy==1.17.0=py37h19fb1c0_0
> >   but it don't work
> > 
> > Running on:
> > 
> > * [ -] CPU
> > * [ ]   GPU
> > 
> > Interface:
> > 
> > * [ ]   C++
> > * [ -] Python
> > 
> > # Reproduction instructions
> > File "e:/BaiduNetdiskDownload/jingdong_NLP/lecture_code/Assignment3-1/Assignment3-1/retrieval/hnsw_faiss.py", line 23, in
> > ```
> > import faiss
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss__init__.py", line 18, in
> > from .loader import *
> > File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\loader.py", line 65, in
> > from .swigfaiss import *
> > File "D:\Users\Nandu\AppData\Local\Continuum\anaconda3\envs\pytorch-gpu\lib\site-packages\faiss\swigfaiss.py", line 13, in
> > from . import _swigfaiss
> > ImportError: DLL load failed: 找不到指定的模块。
> 
> pip uninstall faiss-cpu
> 
> Install new version
> 
> pip install faiss-cpu
> 
> It will fix the error now

------I think there's something wrong with VS Code,because I can run "import faiss" in jupyternotebook but it don't work in VS Code。I have'nt fix it yet。

**Comment by Nandu-H** [comment]
> Hi, it's not fixed (as of faiss_cpu-1.7.1.post2-cp38-cp38-win_amd64.whl), and it's a big problem known on forums like StackOverflow. It's preventing windows users from working with faiss, unfortunately.
> 
> ```python
> C:\ProgramData\Anaconda3\lib\site-packages\faiss\swigfaiss.py in <module>
>      11 # Import the low-level C/C++ module
>      12 if __package__ or "." in __name__:
> ---> 13     from . import _swigfaiss
>      14 else:
>      15     import _swigfaiss
> 
> ImportError: DLL load failed while importing _swigfaiss: The specified module could not be found.
> ```
> 
> faiss-gpu is not installing on Win at all.

---- I think there's something wrong with VS Code,because I can run "import faiss" in jupyternotebook but it don't work in VS Code。I have'nt fix it yet。

**Comment by Nandu-H** [comment]
> Hi, it's not fixed (as of faiss_cpu-1.7.1.post2-cp38-cp38-win_amd64.whl), and it's a big problem known on forums like StackOverflow. It's preventing windows users from working with faiss, unfortunately.
> 
> ```python
> C:\ProgramData\Anaconda3\lib\site-packages\faiss\swigfaiss.py in <module>
>      11 # Import the low-level C/C++ module
>      12 if __package__ or "." in __name__:
> ---> 13     from . import _swigfaiss
>      14 else:
>      15     import _swigfaiss
> 
> ImportError: DLL load failed while importing _swigfaiss: The specified module could not be found.
> ```
> 
> faiss-gpu is not installing on Win at all.

maybe you can see issue #2107 , it may help you figure out this problem i think

**Comment by Dontoronto** [comment]
I previously installed faiss-cpu with this command ("-c pytorch" as source was also not working):

`conda install faiss-cpu --solver=libmamba  `

this wasn't working so i uninstalled it with:

` conda uninstall faiss-cpu   `

the solution which was working was:

`conda install -c conda-forge faiss  `
